warrior/Framework/Utils/email_utils.py

```diff
@@ -165,14 +165,24 @@ def compose_send_email(exec_type, abs_filepath, logs_dir, results_dir, result,
                 (1) per_execution(default)
                 (2) first_failure
                 (3) every_failure
+        7. compress - specifies if the result has to be compressed and emailed.
+           Supported options: Yes/No(default)
     """
     resultconverted = {"True": "Pass", "False": "Fail", "ERROR": "Error",
                        "EXCEPTION": "Exception"}.get(str(result))
     subject = str(resultconverted)+": "+file_Utils.getFileName(abs_filepath)
     body = construct_mail_body(exec_type, abs_filepath, logs_dir, results_dir)
     report_attachment = results_dir + os.sep + \
-        file_Utils.getNameOnly(file_Utils.getFileName(abs_filepath)) + ".html"
-
+               file_Utils.getNameOnly(file_Utils.getFileName(abs_filepath)) 
+    warrior_tools_dir = Tools.__path__[0]+os.sep+'w_settings.xml'
+    element = ET.parse(warrior_tools_dir)
+    setting_elem = element.find("Setting[@name='mail_to']")
+    if setting_elem is not None:
+        compress = setting_elem.get("compress")
+        if "Yes" in compress:
+            report_attachment += ".zip"
+        else:
+            report_attachment += ".html"
```

Comment:
I suggest to perform this operation inside 'set_params_send_email' to avoid parsing 'w_settings.xml' multiple times in different places. Construction of 'files' dict can be moved to 'get_email_params' method.

compress attribute should be case insensitive & can accept 'y/n' as well. Use equality to check in if condition.

Compression can happen here instead of html_results_class.py since zip is only for sending email and we should still print the path of the html for the user.

Reply:
Addressed the comments.